demos/TicTacToe/Classes/MainMenuScene.cpp

```diff
@@ -22,33 +76,345 @@ bool MainMenuScene::init() {
+               password_text_field_size.height)};
+
+  // Creating a text field border and adding it around the text field.
+  auto password_text_field_border = DrawNode::create();
+  password_text_field_border->drawPolygon(
+      password_border_corners, 4, Color4F(0, 0, 0, 0), 1, Color4F::WHITE);
+  // Creating a text field to enter the user's password.
+  password_text_field = cocos2d::TextFieldTTF::textFieldWithPlaceHolder(
+      "enter an password", password_text_field_size, TextHAlignment::LEFT,
+      "Arial", password_font_size);
+  password_text_field->setPosition(password_text_field_position);
+  password_text_field->setAnchorPoint(Vec2(0, 0));
+  password_text_field->setColorSpaceHolder(Color3B::GRAY);
+  password_text_field->setSecureTextEntry(true);
+  password_text_field->setDelegate(this);
+
+  auto password_text_field_touch_listener =
+      EventListenerTouchOneByOne::create();
+
+  password_text_field_touch_listener->onTouchBegan =
+      [this](cocos2d::Touch* touch, cocos2d::Event* event) -> bool {
+    if (previous_state != current_state || current_state != kAuthState) {
+      return true;
+    }
+    auto bounds = event->getCurrentTarget()->getBoundingBox();
+    auto point = touch->getLocation();
+    if (bounds.containsPoint(point)) {
+      // Show the on screen keyboard and places character inputs into the text
+      // field.
+      auto str = password_text_field->getString();
+      auto text_field = dynamic_cast<TextFieldTTF*>(event->getCurrentTarget());
+      text_field->setCursorEnabled(true);
+      text_field->attachWithIME();
+    } else {
+      auto text_field = dynamic_cast<TextFieldTTF*>(event->getCurrentTarget());
+      text_field->setCursorEnabled(false);
+      text_field->detachWithIME();
+    }
+
+    return true;
+  };
+
+  auth_background->addChild(password_text_field, 1);
+  auth_background->addChild(password_text_field_border, 1);
+
+  // Attaching the touch listener to the text field.
+  Director::getInstance()
+      ->getEventDispatcher()
+      ->addEventListenerWithSceneGraphPriority(
+          password_text_field_touch_listener, password_text_field);
+  // Creating the login button and giving it a position, anchor point and
+  // touch_listener.
+  auto login_button = Sprite::create(kLoginButtonImage);
+  login_button->setPosition(90, 120);
+  login_button->setAnchorPoint(Vec2(0, 0));
+  login_button->setContentSize(Size(200, 75));
   // Create a button listener to handle the touch event.
-  auto create_button_touch_listener = EventListenerTouchOneByOne::create();
-  // Setting the onTouchBegan event up to a lambda tha will replace the MainMenu
-  // scene with a TicTacToe scene.
-  create_button_touch_listener->onTouchBegan = [](Touch* touch,
-                                                  Event* event) -> bool {
+  auto login_button_touch_listener = EventListenerTouchOneByOne::create();
+  // Setting the onTouchBegan event up to a lambda tha will replace the
```

Comment:
Suggestion: "Transition from MainMenu scene to TicTacToe on button press, and pass in the login_text_field string."